### PR TITLE
ui: Added sidebar Tree View that displays problem lists. 

### PR DESCRIPTION
**Pull request by l1xiangyi**
Set up UI, navigation in `main.tsx`. Modified `coding.tsx` to receive rout params. Created Chevron svg and related animation.

### CODE REVIEW

File: src/api/navigation-types.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import {ParamListBase} from '@react-navigation/native';
+import {NativeStackNavigationProp} from '@react-navigation/native-stack';
+
+export interface RootStackParamList extends ParamListBase {
+  main: undefined;
+  coding: {titleSlug: string, frontendQuestionId: string};
```
Comment (JumjieYang) [inline src/api/navigation-types.ts:6]: Need type

File: src/api/navigation-types.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+}
+
+export type MainScreenNavigationProp = NativeStackNavigationProp<RootStackParamList, 'main'>;
+export type CodingRouteNavigationProp = {params: {titleSlug: string, frontendQuestionId: string}};
```
Comment (JumjieYang) [inline src/api/navigation-types.ts:10]: Need refactor

File: src/pages/coding.tsx [hunk 3]
```diff
@@ -1,13 +1,33 @@
...
+} from 'react-native';
+import {useNavigation, useRoute} from '@react-navigation/native';
+import {RouteProp} from '@react-navigation/core';
+
+import RenderHTML from 'react-native-render-html';
+
+import {RootStackParamList} from 'api/navigation-types';
 import ApiCaller from '../api/apicaller';
-import {ProblemDetail} from '../api/interfaces';
+import {ProblemDetail} from 'api/interfaces';
```
Comment (JumjieYang) [inline src/pages/coding.tsx:19]: What's the diff

File: src/pages/coding.tsx [hunk 4]
```diff
@@ -1,13 +1,33 @@
...
 import ACSvg from '../svg/ac';
 import RejSvg from '../svg/rej';
 
+type CodingRouteNavigationProp = RouteProp<RootStackParamList, 'coding'>;
+
 export const CodingPage: React.FC = () => {
   const navigator: any = useNavigation();
+  const route = useRoute<CodingRouteNavigationProp>();
+  const titleSlug = route.params.titleSlug;
+  console.log(titleSlug);
```
Comment (JumjieYang) [inline src/pages/coding.tsx:29]: Why console.log

File: src/pages/coding.tsx [hunk 5]
```diff
@@ -18,16 +38,22 @@ export const CodingPage: React.FC = () => {
...
-  // );
+
+  // logic for rendering html strings in the problem details part.
+  const {width} = useWindowDimensions();
+  const theme = useColorScheme();
+  const tagStyles = {
+    body: {
+      color: theme === 'dark' ? 'white' : 'black',
+    },
+  };
```
Comment (JumjieYang) [inline src/pages/coding.tsx:49]: Why not use tailwind

File: src/pages/main.tsx [hunk 6]
```diff
@@ -1,22 +1,87 @@
...
   const navigator: any = useNavigation();
 
   const fetchQuestions = async () => {
-    const questions = await ApiCaller.getInstance().getProblems(
-      '',
-      {difficulty: 'EASY'},
-      50,
-      0,
-    );
+    const questions = await ApiCaller.getInstance().getProblems('', {}, 50, 0);
```
Comment (JumjieYang) [inline src/pages/main.tsx:84]: Need dynamic update

File: src/pages/main.tsx [hunk 7]
```diff
@@ -42,13 +107,34 @@ export const MainPage: React.FC = () => {
...
-        <Pressable
-          className="absolute ml-[2.344vw] mt-[52.188vh] w-[10.33vw] h-[4.199vh] rounded-[20px] border-[#FFAA44] border-2 justify-between items-center"
-          onPress={() => navigator.navigate('coding')}>
-          <Text className="pt-[0.6vh] text-[#FFAA44] text-[1.47vw] leading-tight">
-            Coding
-          </Text>
-        </Pressable>
+
+        <ScrollView className="flex flex-col max-h-[75vh] p-6">
+          {/* <TreeView /> */}
```
Comment (JumjieYang) [inline src/pages/main.tsx:112]: Why empty block

File: src/pages/main.tsx [hunk 8]
```diff
@@ -42,13 +107,34 @@ export const MainPage: React.FC = () => {
...
+
+          <TreeNode label="LeetCode">
+            <TreeNode label="All">
+              {questionData?.questions.slice(0, 10).map(ele => (
+                <QuestionNode question={ele} />
+              ))}
+            </TreeNode>
+            <TreeNode label="Company">
+              <Text>Yeah</Text>
```
Comment (JumjieYang) [inline src/pages/main.tsx:121]: ?

File: src/pages/main.tsx [hunk 9]
```diff
@@ -42,13 +107,34 @@ export const MainPage: React.FC = () => {
...
+            </TreeNode>
+
+            <TreeNode label="Difficulty">
+              <TreeNode label="Easy">
+                <Text>Yeah</Text>
```
Comment (JumjieYang) [inline src/pages/main.tsx:126]: ?

File: src/pages/main.tsx [hunk 10]
```diff
@@ -42,13 +107,34 @@ export const MainPage: React.FC = () => {
...
+              </TreeNode>
+              <TreeNode label="Medium">
+                <Text>Yeah</Text>
```
Comment (JumjieYang) [inline src/pages/main.tsx:129]: ?

File: src/pages/main.tsx [hunk 11]
```diff
@@ -42,13 +107,34 @@ export const MainPage: React.FC = () => {
...
+              </TreeNode>
+              <TreeNode label="Hard">
+                <Text>Yeah</Text>
```
Comment (JumjieYang) [inline src/pages/main.tsx:132]: ?

File: src/svg/chevron.tsx [hunk 12]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import {useWindowDimensions} from 'react-native';
+import {SvgXml} from 'react-native-svg';
+
+const Chevron = () => {
+  const {height, width} = useWindowDimensions();
+
+  const computedWidth = (10 * width) / 1265;
+  const computedHeight = (10 * height) / 1024;
+  const bookXML = `
```
Comment (JumjieYang) [inline src/svg/chevron.tsx:10]: not book

File: tsconfig.json [hunk 13]
```diff
@@ -1,3 +1,6 @@
 {
-  "extends": "@tsconfig/react-native/tsconfig.json"
+  "extends": "@tsconfig/react-native/tsconfig.json",
+  "compilerOptions": {
+    "baseUrl": "src"
+  }
```
Comment (JumjieYang) [inline tsconfig.json:5]: Why need to modify tsconfig

**Review by JumjieYang — CHANGES_REQUESTED** [review]

File: src/pages/coding.tsx (hunk 5 above)
Comment (l1xiangyi) [reply src/pages/coding.tsx:49]: RenderHTML doesn't have className prop.

**Pull request closed without merging**